opentelemetry-contrib/src/trace/propagator/datadog.rs

```diff
@@ -0,0 +1,154 @@
+use opentelemetry::{
+    propagation::{text_map_propagator::FieldIter, Extractor, Injector, TextMapPropagator},
+    trace::{
+        SpanContext, SpanId, TraceContextExt, TraceId, TraceState, TRACE_FLAG_NOT_SAMPLED,
+        TRACE_FLAG_SAMPLED,
+    },
+    Context,
+};
+
+const DATADOG_TRACE_ID_HEADER: &'static str = "x-datadog-trace-id";
+const DATADOG_PARENT_ID_HEADER: &'static str = "x-datadog-parent-id";
+const DATADOG_SAMPLING_PRIORITY_HEADER: &'static str = "x-datadog-sampling-priority";
+// TODO: Implement origin propagation once the Context API is stable
+// const HTTP_HEADER_ORIGIN: &'static str = "x-datadog-origin";
```

Comment:
I need to check if this is stable now (or what it's for for that matter), and how this would integrate with OpenTelemetry.

Might be worth just pulling this out into an issue to track for later.

Reply:
Yeah could open an issue for this one